client/collection_graph_test.go

```diff
@@ -0,0 +1,39 @@
+package client
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCollectionGraph(t *testing.T) {
+	t.Run("Get collection graph", func(t *testing.T) {
+		expected := CollectionGraph{
+			Revision: 1,
+			Groups: map[string]map[string]string{
+				"1": {
+					"1":  "read",
+					"10": "write",
+					"9":  "none",
+				},
+				"4": {
+					"1":  "none",
+					"10": "none",
+					"9":  "none"},
+			},
+		}
+		httpMethod := http.MethodGet
+		svr := server("/api/collection/graph", httpMethod, expected)
+		defer svr.Close()
+
+		c := Client{
+			BaseURL:    svr.URL,
+			HTTPClient: &http.Client{},
+		}
+
+		pg, _ := c.GetCollectionGraph()
+
+		assert.Equal(t, expected, pg)
+	})
```

Comment:
Further, on testing - Right now, we are only unit-testing the client because when I picked up this provider, I had to learn a lot of concepts.

In future, as the usage grows, we will add [acceptance tests](https://www.terraform.io/plugin/sdkv2/testing/acceptance-tests) covering the resource code using the client as well. Just FYI